src/Compilers/CSharp/Test/Emit2/CodeGen/CodeGenRefReadonlyParameterTests.cs

```diff
@@ -0,0 +1,474 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests.CodeGen;
+
+public class CodeGenRefReadonlyParameterTests : CSharpTestBase
+{
+    private const string RequiresLocationAttributeName = "RequiresLocationAttribute";
+
+    [Fact]
+    public void Method()
```

Comment:
Let's also verify that the attribute was initially missing and got added to the compilation (appears in PE but not in source).
Let's add a similar test with RequiresLocationAttribute that exists.